src/TestFramework/Coverage/CoverageChecker.php

```diff
@@ -0,0 +1,194 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\TestFramework\Coverage;
+
+use Composer\XdebugHandler\XdebugHandler;
+use function count;
+use const DIRECTORY_SEPARATOR;
+use function extension_loaded;
+use function file_exists;
+use function implode;
+use Infection\TestFramework\Coverage\XmlReport\PhpUnitXmlCoverageFactory;
+use const PHP_EOL;
+use const PHP_SAPI;
+use function Safe\preg_match;
+use function Safe\sprintf;
+use Webmozart\PathUtil\Path;
+
+/**
+ * @internal
+ */
+final class CoverageChecker
+{
+    private $skipCoverage;
+    private $skipInitialTests;
+    private $initialTestPhpOptions;
+    private $coveragePath;
+    private $jUnitPath;
+
+    public function __construct(
+        bool $skipCoverage,
+        bool $skipInitialTests,
+        string $initialTestPhpOptions,
+        string $coveragePath,
+        string $jUnitPath
+    ) {
+        $this->skipCoverage = $skipCoverage;
+        $this->skipInitialTests = $skipInitialTests;
+        $this->initialTestPhpOptions = $initialTestPhpOptions;
+        $this->coveragePath = $coveragePath;
+        $this->jUnitPath = $jUnitPath;
+    }
+
+    public function checkCoverageRequirements(): void
+    {
+        if ($this->skipInitialTests && !$this->skipCoverage) {
+            throw new CoverageNotFound(
+                'The initial test suite run is being skipped. The XML and JUnit coverage '
+                . 'reports need to be provided with the "--coverage" option'
+            );
+        }
+
+        if (!$this->skipCoverage && !$this->hasCoverageGeneratorEnabled()) {
+            throw new CoverageNotFound(<<<TXT
+Coverage needs to be generated but no code coverage generator (pcov, phpdbg or xdebug) has been detected. Please either:
+- Enable pcov and run infection again
+- Use phpdbg, e.g. `phpdbg -qrr infection`
+- Enable Xdebug and run infection again
+- Use the "--coverage" option with path to the existing coverage report
+- Enable the code generator tool for the initial test run only, e.g. with `--initial-tests-php-options -d zend_extension=xdebug.so`
+TXT
+            );
+        }
+    }
+
+    public function checkCoverageExists(): void
+    {
+        $coverageIndexFilePath = Path::canonicalize(
+            $this->coveragePath . DIRECTORY_SEPARATOR . PhpUnitXmlCoverageFactory::COVERAGE_INDEX_FILE_NAME
+        );
+
+        if (!file_exists($coverageIndexFilePath)) {
+            throw new CoverageNotFound(sprintf(
+                'Could not find the file "%s". Please ensure that the XML coverage report has been '
+                . 'properly generated at the right place. If using PHPUnit for example, the option '
+                . 'for the path given is "--coverage-xml=%s"',
+                $coverageIndexFilePath,
+                $this->coveragePath . '/coverage-xml'
+            ));
+        }
+
+        if (!file_exists($this->jUnitPath)) {
```

Comment:
It won't work for PHPSpec (`TestFrameworkAdapter::hasJUnitReport()`).

Should be tested somehow to avoid these mistakes.